explorer/nodewatch/tests/test_RoutesFacade.py

```diff
@@ -433,26 +443,43 @@ def test_can_generate_nodes_json_filtered_ssl(self):
 			[2, 3],
 			list(map(lambda descriptor: descriptor['roles'], node_descriptors)))
 
-	def test_can_generate_nodes_json_filtered_order_random_limit_2(self):
+	def test_can_generate_nodes_json_filtered_order_random_subset(self):
 		# Arrange:
 		facade = SymbolRoutesFacade(SymbolNetwork.MAINNET, '<symbol_explorer>')
 		facade.reload_all(Path('tests/resources'), True)
 
 		# Act: select 2 nodes with order random
 		node_descriptors = facade.json_nodes(limit=2, order='random')
 
-		# returns all nodes
+		# Assert:
 		all_node_descriptors = facade.json_nodes()
 
 		full_node_names = list(map(lambda descriptor: descriptor['name'], all_node_descriptors))
 		random_node_names = list(map(lambda descriptor: descriptor['name'], node_descriptors))
 
-		# Assert: spot check names
 		self.assertEqual(2, len(node_descriptors))
+		self.assertEqual(2, len(set(random_node_names)))
 		for name in random_node_names:
 			self.assertIn(name, full_node_names)
 
-	def test_can_generate_node_json_given_main_public_key(self):
+	def test_can_generate_nodes_json_filtered_limit_5(self):
+		# Arrange:
+		facade = SymbolRoutesFacade(SymbolNetwork.MAINNET, '<symbol_explorer>')
+		facade.reload_all(Path('tests/resources'), True)
+
+		# Act:
+		node_descriptors = facade.json_nodes(limit=5)
+
+		# Assert: spot check names
+		self.assertEqual(5, len(node_descriptors))
+		self.assertEqual(
+			['Allnodes250', 'Allnodes251', 'Apple', 'Shin-Kuma-Node', 'ibone74'],
+			list(map(lambda descriptor: descriptor['name'], node_descriptors)))
+		self.assertEqual(
+			[2, 2, 7, 3, 3],
+			list(map(lambda descriptor: descriptor['roles'], node_descriptors)))
+
+	def can_find_known_node_by_main_public_key(self):  # pylint: disable=invalid-name
```

Comment:
test need to start with test_ prefix or they will not be detected as tests  (x4)